src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
         }
       }
 
-      isFetching.value = false;
       quote.value = data;
-      runOnJS(updateQuoteStore)(data);
 
       if (!data || (data as QuoteError)?.error) {
         return;
       }
 
       if (inputAmount) {
-        const price = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+        const price = inputPrice || inputNativePrice.value;
         inputValues.modify(prev => {
+          updateNativePriceForAsset({ price, type: 'inputAsset' });
           return {
             ...prev,
-            inputAmount,
+            inputAmount: Number(inputAmount),
```

Comment:
wondering why we're doing an extra cast to `Number` when `inputAmount` is already a number type